plugins/inputs/http/http.go

```diff
@@ -116,13 +136,69 @@ func (h *HTTP) Init() error {
 
 	h.client = client
 
+	if h.CookieAuthURL != "" {
+		// cookie auth defaults
+		renewalInterval := defaultCookieAuthRenewal
+		if h.CookieAuthRenewal != 0 {
+			renewalInterval = time.Duration(h.CookieAuthRenewal)
+		}
+		if h.CookieAuthMethod == "" {
+			h.CookieAuthMethod = defaultCookieAuthMethod
+		}
+		// add cookie jar to HTTP client
+		if h.client.Jar, err = cookiejar.New(nil); err != nil {
+			return err
+		}
+		// start auth ticker
+		if authErr := h.startCookieAuth(renewalInterval); authErr != nil {
+			return authErr
+		}
+	}
+
 	// Set default as [200]
 	if len(h.SuccessStatusCodes) == 0 {
 		h.SuccessStatusCodes = []int{200}
 	}
 	return nil
 }
 
+func (h *HTTP) startCookieAuth(interval time.Duration) error {
+	// continual auth ticker
+	ticker := time.NewTicker(interval)
+	go func() {
+		for range ticker.C {
+			_ = h.doCookieAuth()
+		}
+	}()
+
+	// initial auth will immediately error out the Init() if auth fails
+	return h.doCookieAuth()
+}
+
+func (h *HTTP) doCookieAuth() error {
+	request, err := h.newRequest(h.CookieAuthURL, h.CookieAuthMethod, h.CookieAuthBody)
```

Comment:
I don't think reusing the existing http-client will work here. Let me explain why: in `newRequest()` you are setting the parameters for the request including (shortened)
```golang
	bearer := "Bearer " + strings.Trim(string(token), "\n")
	request.Header.Set("Authorization", bearer)

	request.Header.Set("Content-Encoding", "gzip")

	for k, v := range h.Headers {
		request.Host = v
		request.Header.Add(k, v)
	}

	request.SetBasicAuth(h.Username, h.Password)
```
Now you use a completely different URL to issue the request, reusing the data originally specified for one URL on another URL. Not only, these setting might potentially settings between the URL but you also **leak sensible data** to a (probably) different server than initially intended by the user.

This being said, I think you need to setup a second http-client to perform the cookie-authentication, potentially also allowing the user to specify encoding, credentials, headers,...

Reply:
Yes, I can do that.
The original use case was for a single host, so that's the direction I took since the URL slice is already "bound" to a single HTTP Basic config (user/pass) in the same config.

Reply:
TBH, I also don't like the ability to specify multiple URLs as it calls for errors for exactly the reason you mentioned. However, we should not make things worse by adding another issue. :-)